src/components/stepforms/JoinStepForm.vue

```diff
@@ -77,14 +82,14 @@ export default class JoinStepForm extends BaseStepForm<JoinStep> {
   joinColumns = JoinColumns;
   joinTypes: JoinStep['type'][] = joinTypes;
 
-  get rightPipeline(): object {
+  get rightPipeline(): DropdownOption {
     return {
-      label: this.editedStep.right_pipeline,
-      trackBy: this.editedStep.right_pipeline,
+      label: this.editedStep.right_pipeline as string,
+      trackBy: this.editedStep.right_pipeline as string,
```

Comment:
Yes, this is quite unconfortable as  we do not have different types for steps with pipeline references and steps with compete pipelines. I think we should create an issue in that.